viskores/exec/SplineEvaluateRectilinearGrid.h

```diff
@@ -0,0 +1,259 @@
+        C2[jj] = this->CubicInterpolateNonUniform(
+          x0, x1, x2, x3, P2d[jj * 4 + 0], P2d[jj * 4 + 1], P2d[jj * 4 + 2], P2d[jj * 4 + 3], x);
+      }
+      // 4) bicubic along Y on C2 → final
+      auto y0 = this->AxisPortals[1].Get(iv - 1);
+      auto y1 = this->AxisPortals[1].Get(iv);
+      auto y2 = this->AxisPortals[1].Get(iv + 1);
+      auto y3 = this->AxisPortals[1].Get(iv + 2);
+      value = this->CubicInterpolateNonUniform(y0, y1, y2, y3, C2[0], C2[1], C2[2], C2[3], y);
+      return viskores::ErrorCode::Success;
+    }
+
+    viskores::FloatDefault P[4 * 4 * 4];
+    for (viskores::Id kk = 0; kk < 4; ++kk)
+    {
+      viskores::Id k = viskores::Clamp(iw - 1 + kk, 0, this->NumZ - 1);
+      for (viskores::Id jj = 0; jj < 4; ++jj)
+      {
+        viskores::Id j = viskores::Clamp(iv - 1 + jj, 0, this->NumY - 1);
+        for (viskores::Id ii = 0; ii < 4; ++ii)
+        {
+          viskores::Id i = viskores::Clamp(iu - 1 + ii, 0, this->NumX - 1);
+          auto pIndex = (kk * 4 + jj) * 4 + ii;
+          auto dIndex = (k * this->NumY + j) * this->NumX + i;
+          P[pIndex] = this->Field.Get(dIndex);
+        }
+      }
+    }
+
+    // interpolate in X for each (kk,jj) → Cbuf[16]
+    viskores::FloatDefault Cbuf[4 * 4];
+    for (int kk = 0; kk < 4; ++kk)
+      for (int jj = 0; jj < 4; ++jj)
+      {
+        auto x0 = this->AxisPortals[0].Get(iu - 1);
+        auto x1 = this->AxisPortals[0].Get(iu);
+        auto x2 = this->AxisPortals[0].Get(iu + 1);
+        auto x3 = this->AxisPortals[0].Get(iu + 2);
+        auto p0 = P[(kk * 4 + jj) * 4 + 0];
+        auto p1 = P[(kk * 4 + jj) * 4 + 1];
+        auto p2 = P[(kk * 4 + jj) * 4 + 2];
+        auto p3 = P[(kk * 4 + jj) * 4 + 3];
+        Cbuf[kk * 4 + jj] = this->CubicInterpolateNonUniform(x0, x1, x2, x3, p0, p1, p2, p3, x);
+      }
+
+    // interpolate in Y → D[4]
+    viskores::FloatDefault D[4];
+    for (int kk = 0; kk < 4; ++kk)
+    {
+      auto y0 = this->AxisPortals[1].Get(iv - 1);
+      auto y1 = this->AxisPortals[1].Get(iv);
+      auto y2 = this->AxisPortals[1].Get(iv + 1);
+      auto y3 = this->AxisPortals[1].Get(iv + 2);
+      D[kk] = this->CubicInterpolateNonUniform(
+        y0, y1, y2, y3, Cbuf[kk * 4 + 0], Cbuf[kk * 4 + 1], Cbuf[kk * 4 + 2], Cbuf[kk * 4 + 3], y);
+    }
+
+    // interpolate in Z
+    auto z0 = this->AxisPortals[2].Get(iw - 1);
+    auto z1 = this->AxisPortals[2].Get(iw);
+    auto z2 = this->AxisPortals[2].Get(iw + 1);
+    auto z3 = this->AxisPortals[2].Get(iw + 2);
+    value = this->CubicInterpolateNonUniform(z0, z1, z2, z3, D[0], D[1], D[2], D[3], z);
+
+    return viskores::ErrorCode::Success;
+  }
+
+private:
+  VISKORES_EXEC viskores::Id FindIndex(const AxisPortalType& axis,
+                                       const viskores::Id& N,
+                                       viskores::FloatDefault val) const
+  {
+    // 1) Binary search for the largest index i with coords[i] <= val
```

Comment:
I realize I'm being pedantic here, but what if the coordinates are not in increasing order? What if they are monotonically decreasing? That might happen to work around a left-handed/right-handed grid issue.

Perhaps not worth doing, and it could be done in a separate PR.